gtdynamics/pandarobot/ikfast/PandaIKFast.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file  PandaIKFast.h
+ * @brief Wrapper for the generated IKFast source files for the panda robot.
+ * @author Frank Dellaert
+ * @author Antoni Jubes
+ */
+
+#include "PandaIKFast.h"
+
+//----------------------------------------------------------------------------//
+
+#define IKFAST_HAS_LIBRARY  // Build IKFast with API functions
+#define IKFAST_NO_MAIN      // Don't include main() from IKFast
+#define IKFAST_NAMESPACE panda_internal
+
+#define IK_VERSION 61
+#include <gtdynamics/pandarobot/ikfast/ikfast.h>
+
+//----------------------------------------------------------------------------//
+
+#include <gtsam/base/Vector.h>
+#include <gtsam/geometry/Pose3.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <iostream>
+#include <vector>
+
+namespace gtdynamics {
+
+PandaIKFast::PandaIKFast() {}
+
+gtsam::Pose3 PandaIKFast::forward(const gtsam::Vector7& joint_values) {
```

Comment:
In cpp you should use `using` as much as possible :-) at top of file, one line per gtsam class, e.g.,

```
using gtsam::Pose3;
```

still no blanket namespace using….

Reply:
Ooh, finally got it. I did not understand the reason earlier, just thought that it was arbitrary. Now it makes sense :-)